bin/proving-service/src/proxy/health_check.rs

```diff
@@ -38,49 +30,16 @@ impl BackgroundService for LoadBalancerState {
                 // Create a new spawn to perform the health check
                 let span = debug_span!("proxy:health_check");
                 let _guard = span.enter();
+                {
+                    let mut workers = self.workers.write().await;
 
-                let mut workers = self.workers.write().await;
-                let initial_workers_len = workers.len();
-
-                // Perform health checks on workers and retain healthy ones
-                let healthy_workers = self.check_workers_health(workers.iter_mut()).await;
-
-                // Update the worker list with healthy workers
-                *workers = healthy_workers;
-
-                // Update the worker count and worker unhealhy count metrics
-                WORKER_COUNT.set(workers.len() as i64);
-                let unhealthy_workers = initial_workers_len - workers.len();
-                WORKER_UNHEALTHY.inc_by(unhealthy_workers as u64);
-
+                    // Perform status checks on workers and retain ready ones
+                    self.check_workers_status(workers.iter_mut()).await;
+                }
```

Comment:
Do we actually need to have `LoadBalancerState::check_worker_status()` method? Could we just do something like:
```rust
let mut workers = self.workers.write().await;

for worker in workers {
    worker.check_status(self.supported_prover_type).await;
}
```

Also, if I'm understanding correctly, this will currently lock the workers while health checks are happening. If so, then no requests could be served during this time, right? This is something we'll need to address, but probably in a follow-up PR.